src/dsp_tools/utils/connection_live.py

```diff
@@ -271,7 +271,7 @@ def _try_network_action(self, params: RequestParameters) -> Response:
                 return response
             elif "v2/authentication" in params.url and response.status_code == HTTP_UNAUTHORIZED:
                 raise BadCredentialsError("Bad credentials")
-            elif "OntologyConstraintException" in response.text:
+            elif any(x in response.text for x in ["OntologyConstraintException", "DuplicateValueException"]):
```

Comment:
I'm personally not a big fan of these kinds of logical statements, they simply don't look very readable to me. On the other side, I fully acknowledge that it's exactly the logic we need. How would you feel about extracting `any(x in response.text for x in ["OntologyConstraintException", "DuplicateValueException"])` into a little function `_isNotRetryable()` or `_isNoRetryException()` or `_noRetryNeeded()` or something like that, so on the caller side we would have something more speaking, and the implementation detail is hidden away?